src/parse_json_for_plots.py

```diff
@@ -0,0 +1,241 @@
+        blosc_compressors
+    )
+    benchmark_df.loc[~benchmark_df["params.gzip_level"].isna(), "compressor"] = "gzip"
+    benchmark_df.loc[~benchmark_df["params.zstd_level"].isna(), "compressor"] = "zstd"
+
+    # remove un-needed columns
+    stats_cols = [col for col in benchmark_df if col.startswith("stats")]
+    benchmark_df = benchmark_df[
+        [
+            "group",
+            "compressor",
+            "compression_level",
+            "compression_ratio",
+            "params.chunk_size",
+        ]
+        + stats_cols
+    ]
+    benchmark_df = benchmark_df.rename(columns={"params.chunk_size": "chunk_size"})
+
+    return benchmark_df
+
+
+def get_benchmarks_dataframe(
+    json_paths: tuple[Path], package_ids: tuple[str]
+) -> pd.DataFrame:
+    """Combine multiple pytest-benchmark json results into a single dataframe"""
+
+    benchmark_dfs = []
+    for json_path, id in zip(json_paths, package_ids):
+        benchmark_df = prepare_benchmarks_dataframe(load_benchmarks_json(json_path))
+        benchmark_df.insert(0, "package", id)
+        benchmark_dfs.append(benchmark_df)
+
+    return pd.concat(benchmark_dfs, ignore_index=True)
+
+
+def plot_scatter_benchmarks(
+    df: pd.DataFrame, group: str, palette_dict=None, path_to_file=None
+):
+    x_label = group.capitalize() + " Time (s)"
+    plt.clf()
+    plt.figure(figsize=(10, 6))
+
+    # Add labels and title
+    plt.xlabel(x_label)
+    plt.ylabel("Compression Ratio")
+    plt.title(f"Compression Ratio vs {x_label} for Different Compression Types")
+
+    sns.scatterplot(
+        data=df,
+        x="write_time",
+        y="compression_ratio",
+        hue="compression_type",
+        style="compression_level",
+        alpha=0.7,
+        palette=palette_dict,
+        s=100,
+    )
+
+    save_plot_as_png(
+        plt,
+        f"data/json/{group}_scatter_{Path(path_to_file).stem}.png",
+    )
+
+
+def plot_errorbars_benchmarks(df, group, palette_dict=None, path_to_file=None):
+    x_label = group.capitalize() + " Time (s)"
+    plt.clf()
+    plt.figure(figsize=(10, 6))
+
+    # Add labels and title
+    plt.xlabel(x_label)
+    plt.ylabel("Compression Ratio")
+    plt.title(f"Compression Ratio vs {x_label} for Different Compression Types")
+
+    sns.boxplot(
+        x="write_time",
+        y="compression_ratio",
+        hue="compression_type",
+        data=df,
+        palette=palette_dict,
+    )
+
+    save_plot_as_png(
+        plt,
+        f"data/json/{group}_errorbars_{Path(path_to_file).stem}.png",
+    )
+
+
+def plot_relplot_subplots_benchmarks(
+    data: pd.DataFrame,
+    group: str,
+    x_axis: str,
+    y_axis: str,
+    hue: str,
+    path_to_file=None,
+):
+    graph = sns.relplot(
+        data=data,
+        x=x_axis,
+        y=y_axis,
+        col=hue,
+        hue=hue,
+        facet_kws=dict(sharex=False, sharey=False),
+    )
+    graph.set_axis_labels(x_axis.capitalize(), f"Mean {group} time (s)")
```

Comment:
This makes the assumption that `y_axis` will be set to the mean time. If `y_axis` is set to anything else, then the axis label will be incorrect.

If you want to keep it as-is, then maybe rename the function to something like `plot_mean_time_with_subplots` and remove the `y_axis` option. Otherwise you will need to process the axis strings and assign the label accordingly e.g.
```
if x_axis.startswith("stats"):
        label = f"{x_axis.split(".")[-1]} {group} time (s)"
```

Same holds true for `plot_relplot_benchmarks` where you're assuming that the x axis is always the mean time.